### PR TITLE
Affliction 10.0.5

### PR DESCRIPTION
**Pull request by dsnam**
### Description

- updated Soul Conduit for all specs to take into account talent rank scaling, and updated Affliction's module to use rapture as the example spender instead of UA.
- added some hasTalent checks for Darkglare so certain components don't try to render when you don't have it talented
- fixed Vile Taint not checking for the right debuff id
- updated Shadow Embrace and Nightfall to take into account stacking and talent scaling
- Added modules for Wrath of Consumption, Inevitable Demise, Pandemic Invocation, and Tormented Crescendo

### Testing

- Test report URL: 
Log showing Wrath and Vile Taint: `report/4TZ7hgKVbw2jk8Pz/30-Mythic+Sennarth,+The+Cold+Breath+-+Kill+(6:41)/Omenity/standard/statistics`
Log showing Inevitable Demise, Pandemic Invocation, Tormented Crescendo: `report/z3qaHpWNC4JKVYjf/18-Mythic+Terros+-+Kill+(5:34)/Dodse/standard/statistics`
- Screenshot(s):
image
image

### CODE REVIEW

File: src/analysis/retail/warlock/affliction/modules/spells/SoulConduit.tsx
Comment (dsnam) [inline src/analysis/retail/warlock/affliction/modules/spells/SoulConduit.tsx:15]: now that this in the class tree it seems like these should be merged and I could just set what "you were refunded enough shards for x more casts of y spender" message shows based on the combatant spec, but then it'd just be one file with a lot going on. not sure which way of organizing it is preferable.

File: src/analysis/retail/warlock/affliction/modules/spells/InevitableDemise.tsx
Comment (dsnam) [inline src/analysis/retail/warlock/affliction/modules/spells/InevitableDemise.tsx:49]: unlike some of the other buffs with stacks I dealt with in this PR like Tormented Crescendo I noticed the event log for Inevitable Demise has a ton of refresh buff events, basically one for every apply/applystack as well. do some buffs just tend to spam the combat log or is that a relic of this one being older?

Comment (emallson) [reply src/analysis/retail/warlock/affliction/modules/spells/InevitableDemise.tsx:49]: some buffs do that, yea

File: src/analysis/retail/warlock/affliction/modules/spells/SoulConduit.tsx
Comment (emallson) [reply src/analysis/retail/warlock/affliction/modules/spells/SoulConduit.tsx:15]: if its easy, generalizing is good. however, i'd rather have 3 simple modules (which can import shared constants) than 1 complicated module that tries to do everything.

File: src/analysis/retail/warlock/affliction/modules/spells/ShadowEmbrace.tsx
Comment (emallson) [inline src/analysis/retail/warlock/affliction/modules/spells/ShadowEmbrace.tsx:34]: did not know that you could use `this` in property initializers outside the constructor. neat

**Review by emallson — APPROVED** [review]
this all looks pretty reasonable, assuming you're not double-counting stacks due to the `refreshbuff` thing you noted.

**Comment by dsnam** [comment]
@emallson I didn't notice that last comment after I saw it got approved so in case you were waiting on an answer to that before merging: yeah I am pretty sure it's handling the stacks accurately, or at least as accurately as the stack count on the applystack event lets it be. I spot checked it with some logs and the overcapped stack metric was the same as the number of refresh events that came through while at max stacks.

On that note though, is there an existing framework to add tests that are like "run this log or fake event sequence and check that x module's metric has y value," or would something like that be too brittle? I see some modules appear to have something like that but the files are quite old and I can't tell if the code's dead. If it's not I'd be happy to add some for this.

**Comment by emallson** [comment]
> On that note though, is there an existing framework to add tests that are like "run this log or fake event sequence and check that x module's metric has y value," or would something like that be too brittle? I see some modules appear to have something like that but the files are quite old and I can't tell if the code's dead. If it's not I'd be happy to add some for this.

we have some test tools, but with the dragonflight revamp a lot of the old test code got killed. it is possible (but a bit of a pain) to spin up analyzers to run events through, and we have *some* infra to download log data to run against, but it frankly needs a bit of a refresh

**Pull request merged**
